core/src/main/scala/org/genivi/sota/core/DevicesResource.scala

```diff
@@ -107,8 +109,36 @@ class DevicesResource(db: Database, client: ConnectivityClient,
     }
   }
 
+  def getDeviceWithStatus(ns: Namespace, uuid: Uuid): Route = {
+    parameters(('status.?(false))) { (includeStatus: Boolean) =>
+      val devices = deviceRegistry.fetchDevice(uuid).flatMap { device =>
+        if (device.namespace == ns) {
+          Future.successful(Seq(device))
+        } else {
+          Future.failed(MissingEntity(classOf[DeviceSearchResult]))
+        }
+      }.recoverWith {
+        case e => Future.failed(MissingEntity(classOf[DeviceSearchResult]))
+      }
+      val devicesWithStatus = if (includeStatus) {
+        DeviceSearch.fetchDeviceStatus(devices) flatMap buildSearchResponse(devices)
+      } else {
+        buildSearchResponse(devices)(Seq.empty)
+      }
+      complete(devicesWithStatus.map(_.headOption))
+    }
+  }
+
+
+  // Use "devices_info" path
+  // Deprecate "devices" path (which is used in device-registry)
   val route =
-    (pathPrefix("devices") & namespaceExtractor) { ns =>
-      (pathEnd & get) { search(ns) }
+    ((pathPrefix("devices_info") | pathPrefix("devices")) & namespaceExtractor) { ns =>
+      (pathEnd & get) { search(ns) } ~
+      (get & extractUuid) { uuid =>
+        ErrorHandler.handleErrors {
```

Comment:
Any particular reason for having the handleErrors here? There is one in WebService that should catch everything from here right?